Cython/Utility/FunctionArguments.c

```diff
@@ -242,8 +242,8 @@ static int __Pyx_ParseOptionalKeywords(
 
     while (1) {
         if (kwds_is_tuple) {
-            if (pos >= PyTuple_GET_SIZE(kwds)) break;
-            key = PyTuple_GET_ITEM(kwds, pos);
+            if (pos >= __Pyx_PyTuple_GET_SIZE(kwds)) break;
+            key = __Pyx_PyTuple_GET_ITEM(kwds, pos);
```

Comment:
The `GET_ITEM` call should rather be guarded by `SAFE_MACROS` and `BORROWED_REFS`.

Reply:
Still thinking about this. In both cases the `BORROWED_REFS` guard looks hard to implement, and isn't necessary for limited API support (where we allow borrowed refs by default).

So I'd rather just do `SAFE_MACROS` at this stage.

I've moved `__Pyx_PyTuple_GET_SIZE` inside `SAFE_MACROS` but I don't think this is the right solution so may come back and change it.

Reply:
I think these should now be handled (at the cost of some complication)